editor/filesystem_dock.cpp

```diff
@@ -1563,10 +1563,32 @@ void FileSystemDock::_try_duplicate_item(const FileOrFolder &p_item, const Strin
 }
 
 void FileSystemDock::_update_resource_paths_after_move(const HashMap<String, String> &p_renames, const HashMap<String, ResourceUID::ID> &p_uids) const {
-	// Update the paths in ResourceUID, so that UIDs remain valid.
-	for (const KeyValue<String, ResourceUID::ID> &pair : p_uids) {
-		if (p_renames.has(pair.key)) {
-			ResourceUID::get_singleton()->set_id(pair.value, p_renames[pair.key]);
+	for (const KeyValue<String, String> &pair : p_renames) {
+		// Update UID path.
+		if (p_uids.has(pair.key)) {
+			ResourceUID::get_singleton()->set_id(p_uids[pair.key], pair.value);
+			continue; // As long as scripts don't have a UID.
```

Comment:
It would be better to explicitly check for a Script type and `continue` only if it's not a script. Otherwise when Script UIDs are added and someone forgets to handle this, it will lead to a bug.

Reply:
How about just removing it then? It's very few function call until the if block which will not execute if it's not a script

Reply:
I think you should still try to avoid calling these function if the file is not a script, unless it's not simple to implement. If you apply my suggestion about type then it's probably fine.